src/main/java/com/github/alinaberlin/security/components/JWTEncoderDecoder.java

```diff
@@ -0,0 +1,87 @@
+package com.github.alinaberlin.security.components;
+
+import com.github.alinaberlin.security.models.User;
+import io.jsonwebtoken.*;
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class JWTEncoderDecoder {
+    private final String secret_key = "mysecretkeyisthesecretkeymysecretkeyisthesecretkey";
+    private long accessTokenValidity = 60*60*1000;
+
+    private final JwtParser jwtParser;
+
+    private final String TOKEN_HEADER = "Authorization";
```

Comment:
should be `private static final`. Also it would be good if you created another class where you would store header names. Something like this:
```
public class HttpHeadersConstants {
    public static final String AUTHORIZATION = "Authorization";
}
```